### ISSUE TITLE
datetime wrong output

### ISSUE THREAD

**Issue by unorthodoxgeek** [mariano/node-db-mysql#15]
I'm trying to get a datetime object from the database, when saving it is done correctly, but when fetching it, it returns a string, and a very odd one.
for a datetime object which is supposed to be something like this: "2011-06-06 08:32:57"
it returns a string with: "1970-01-19T16:51:45.016Z"

**Comment by mariano** [comment]
Could you provide me a proper dump of the table (just as you have it in the DB) together with the JS code so I can test?

**Comment by unorthodoxgeek** [comment]
sql dump (using mysql 5):

CREATE TABLE IF NOT EXISTS `messages` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `sender_id` int(11) DEFAULT NULL,
  `reciever_id` int(11) DEFAULT NULL,
  `title` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL,
  `message` text COLLATE utf8_unicode_ci,
  `read` tinyint(1) DEFAULT '0',
  `reply_to` int(11) DEFAULT NULL,
  `folder` int(11) DEFAULT '0',
  `user_ip` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL,
  `user_agent` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL,
  `referrer` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL,
  `created_at` datetime DEFAULT NULL,
  `updated_at` datetime DEFAULT NULL,
  `chat` tinyint(1) DEFAULT '0',
  `room_id` int(11) NOT NULL,
  PRIMARY KEY (`id`),
  KEY `index_messages_on_reciever_id` (`reciever_id`)
) ENGINE=InnoDB  DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci AUTO_INCREMENT=61 ;

## 

## -- Dumping data for table `messages`

INSERT INTO `messages` (`id`, `sender_id`, `reciever_id`, `title`, `message`, `read`, `reply_to`, `folder`, `user_ip`, `user_agent`, `referrer`, `created_at`, `updated_at`, `chat`, `room_id`) VALUES
(1, 4, NULL, NULL, 'Howdy!', 0, NULL, 0, NULL, NULL, NULL, '2011-06-05 10:05:54', NULL, 1, 2),
(2, 2, NULL, NULL, 'Hey there!', 0, NULL, 0, NULL, NULL, NULL, '2011-06-05 10:06:01', NULL, 1, 2),
(4, 2, NULL, NULL, 'Sup??', 0, NULL, 0, NULL, NULL, NULL, '2011-06-05 10:31:45', NULL, 1, 2),
(5, 2, NULL, NULL, 'You''re back!', 0, NULL, 0, NULL, NULL, NULL, '2011-06-05 10:50:15', NULL, 1, 2),
(6, 4, NULL, NULL, 'yep! I''m back baby!', 0, NULL, 0, NULL, NULL, NULL, '2011-06-05 10:50:25', NULL, 1, 2),
(7, 2, NULL, NULL, 'Howdy!', 0, NULL, 0, NULL, NULL, NULL, '2011-06-05 11:16:07', NULL, 1, 2),
(8, 2, NULL, NULL, 'Howdy!', 0, NULL, 0, NULL, NULL, NULL, '2011-06-05 12:16:47', NULL, 1, 2),
(9, 4, NULL, NULL, 'SUP??', 0, NULL, 0, NULL, NULL, NULL, '2011-06-05 12:17:02', NULL, 1, 2),
(10, 4, NULL, NULL, '\nthis', 0, NULL, 0, NULL, NULL, NULL, '2011-06-05 12:17:37', NULL, 1, 2),
(11, 4, NULL, NULL, '\nis', 0, NULL, 0, NULL, NULL, NULL, '2011-06-05 12:17:39', NULL, 1, 2);

JS:
insertion (working fine):

```
db.query().
            insert('messages', ['room_id', 'sender_id', 'message', 'created_at', 'chat'], [room, sender, message, new Date(), 1]).
            execute();
```

fetching:

```
db.query( {async: false }).
            select('messages.id, messages.sender_id, messages.message, messages.room_id, users.name, messages.created_at').from('messages').
            join({ table: 'users', conditions: "users.id = messages.sender_id" }).
            where('chat = 1 and room_id = ?', [id.id]).
            order('`id` DESC').limit(10).
            execute( function(rows,cols){
                messages = rows;
            });
```

and then using the data in the messages object:

```
$.each(messages.reverse(), function(key, value) {
                //console.log(typeof value.created_at); puts out "string"
                //console.log(value.created_at); puts out something akin to "1970-01-19T16:51:45.016Z"
                addMessage(room.id, value.name, value.message, value.name == now.name, value.created_at);
            });
```

thanks!

**Comment by mariano** [comment]
Ok this is weird. it doesn't seem to happen to me at all. Using your dump, and also adding a new record from within JS, the created_at is an object, and the value is displayed properly.

This is the JS code I'm using:

``` javascript
var mysql = require('db-mysql');
new mysql.Database({
    hostname: 'localhost',
    user: 'root',
    password: 'password',
    database: 'node'
}).connect(function(error, server) {
    if (error) {
        return console.log('ERROR while connecting: ' + error);
    }

    this.query().insert('messages', 
        ['room_id', 'sender_id', 'message', 'created_at', 'chat'],
        [ 2, null, 'New test message', new Date(), 1]).
        execute({ async: false });

    this.query().select('*').from('messages').execute(function(err, rows, columns) {
        if (err) {
            return console.log('QUERY error: ' + error);
        }

        for (var key in rows.reverse()) {
            var value = rows[key];
            console.log(typeof value.created_at);
            console.log(value.created_at);
        }
    });
});
```

The output of this script is (notice the first row is the recently added row):

```
object                                                                                                                                                                                                    
Mon, 06 Jun 2011 12:46:05 GMT                                                                                                                                                                             
object                                                                                                                                                                                                    
Sun, 05 Jun 2011 12:17:39 GMT                                                                                                                                                                             
object                                                                                                                                                                                                    
Sun, 05 Jun 2011 12:17:37 GMT                                                                                                                                                                             
object                                                                                                                                                                                                    
Sun, 05 Jun 2011 12:17:02 GMT                                                                                                                                                                             
object                                                                                                                                                                                                    
Sun, 05 Jun 2011 12:16:47 GMT                                                                                                                                                                             
object                                                                                                                                                                                                    
Sun, 05 Jun 2011 11:16:07 GMT                                                                                                                                                                             
object                                                                                                                                                                                                    
Sun, 05 Jun 2011 10:50:25 GMT                                                                                                                                                                             
object                                                                                                                                                                                                    
Sun, 05 Jun 2011 10:50:15 GMT                                                                                                                                                                             
object                                                                                                                                                                                                    
Sun, 05 Jun 2011 10:31:45 GMT                                                                                                                                                                             
object                                                                                                                                                                                                    
Sun, 05 Jun 2011 10:06:01 GMT                                                                                                                                                                             
object                                                                                                                                                                                                    
Sun, 05 Jun 2011 10:05:54 GMT
```

My mysql version is 5.5.13. The only way to get a string in the DATETIME field is to set the option { cast: false } when creating the query, like so:

``` javascript
this.query({ cast: false })...
```

And even then I get a better output than that you specify:

```
string                                                                                                                                                                                                    
2011-06-06 12:49:13                                                                                                                                                                                       
string                                                                                                                                                                                                    
2011-06-06 12:46:05                                                                                                                                                                                       
string                                                                                                                                                                                                    
2011-06-05 12:17:39                                                                                                                                                                                       
string                                                                                                                                                                                                    
2011-06-05 12:17:37                                                                                                                                                                                       
string                                                                                                                                                                                                    
2011-06-05 12:17:02                                                                                                                                                                                       
string                                                                                                                                                                                                    
2011-06-05 12:16:47                                                                                                                                                                                       
string                                                                                                                                                                                                    
2011-06-05 11:16:07                                                                                                                                                                                       
string                                                                                                                                                                                                    
2011-06-05 10:50:25                                                                                                                                                                                       
string                                                                                                                                                                                                    
2011-06-05 10:50:15                                                                                                                                                                                       
string                                                                                                                                                                                                    
2011-06-05 10:31:45                                                                                                                                                                                       
string                                                                                                                                                                                                    
2011-06-05 10:06:01                                                                                                                                                                                       
string                                                                                                                                                                                                    
2011-06-05 10:05:54
```

Can you show me how exactly you created the Database instance, what mysql version you are on, what version of db-mysql you are using (make sure you use the latest), and what platform you are using?

**Comment by unorthodoxgeek** [comment]
that is really odd... I wasn't using the latest db-mysql version, but even when upgrading the problem remained, the string part is due to a bug in now.js, but even inside the getter function the object which is returned is somewhere in the 1970s, which is insanely odd... mysql version is 5.1.41.
I'm using node.js 0.5.0-pre and now.js 0.5.3

**Comment by mariano** [comment]
Fixed in 76553a0aa3353e08086cfa49f08b8a501cfb0e20

Update to latest db-mysql. [View release announcement](http://nodejsdb.org/2011/06/node-db-drivers-update/)
